monitoring/alerts/spark-rules.yml

```diff
@@ -0,0 +1,98 @@
+groups:
+- name: Spark_Alerts
+  rules:
+    - alert: Spark_Dispatcher_Count
+      expr: count(spark_mesos_cluster_driver_launched) >= 50
+      for: 1m
+      labels:
+        severity: informational
+      annotations:
+        description: 'Spark has reached desired number of dispatchers'
+
+    - alert: Spark_Dispatcher_Count_Decreasing
+      expr: count(spark_mesos_cluster_driver_launched offset 5m) - count(spark_mesos_cluster_driver_launched) > 0
+      for: 1m
+      labels:
+        severity: major
+      annotations:
+        description: 'Spark dispatcher count unexpectedly decreasing'
+
+    - alert: Spark_Job_Count
+      expr: count(spark_mesos_cluster_executor_count) >= 1000
+      for: 1m
+      labels:
+        severity: informational
+      annotations:
+        description: 'Spark has reached desired number of jobs'
+
+    - alert: Spark_Job_Count_Decreasing
+      expr: count(spark_mesos_cluster_executor_count offset 5m) - count(spark_mesos_cluster_executor_count) > 0
+      for: 1m
+      labels:
+        severity: major
+      annotations:
+        description: 'Spark job count unexpectedly decreasing'
+
+    - alert: Spark_CPU_Job
+      expr: count(spark_mesos_cluster_executor_count{task_name=~".*monte.*"}) >= 750
+      for: 1m
+      labels:
+        severity: informational
+      annotations:
+        description: 'Spark has reached desired number of CPU jobs'
+
+    - alert: Spark_GPU_Jobs
+      expr: count(spark_mesos_cluster_executor_count{task_name=~".*image.*"}) >= 10
+      for: 1m
+      labels:
+        severity: informational
+      annotations:
+        description: 'Spark has reached desired number of GPU jobs'
+
+    - alert: Spark_Streaming_Jobs
+      expr: count(spark_mesos_cluster_executor_count{task_name=~".*Kafka.*"}) >= 250
+      for: 1m
+      labels:
+        severity: informational
+      annotations:
+        description: 'Spark has reached desired number of Streaming jobs'
+    
+    - alert: Spark_Dispatcher_Queue_Depth
+      expr: spark_mesos_cluster_driver_waiting > 200
```

Comment:
Is this a global value? Does each dispatcher not have its own queue?

Reply:
Yes, every dispatcher has its own queue so if any reach 200 this alert fires but this is actually a bad expression. This would cause prom to process way more values than it should. Proper alert should be: 
```
expr: count(spark_mesos_cluster_driver_waiting > 200) > 0
``` 
so we count if _any_ dispatcher has a depth greater than 200